services/web/server/src/simcore_service_webserver/studies_dispatcher/_projects.py

```diff
@@ -201,14 +205,37 @@ async def _add_new_project(
     await create_or_update_pipeline(app, user.id, project.uuid, product_name)
 
 
-async def acquire_project_with_viewer(
+async def _project_exists(
+    app: web.Application,
+    user_id,
+    project_uuid: ProjectID,
+):
+    try:
+        await get_project_for_user(
+            app=app,
+            project_uuid=f"{project_uuid}",
+            user_id=user_id,
+            include_state=False,
+        )
+        return True
+    except (ProjectNotFoundError, ProjectInvalidRightsError):
```

Comment:
Just a Note: Be sure that you dont shoot yourself in the foot, the function Name indicates you are simply checking if a project exists but then you also return False if the project exists but a user has wrong or lacking permissions. I find this a bit nisleading

Reply:
your point is valid. This is a temp solution until we have "exists" function... which perhaps it is time to do!